consensus/istanbul/backend/engine.go

```diff
@@ -882,6 +820,62 @@ func (sb *Backend) snapshot(chain consensus.ChainReader, number uint64, hash com
 	return returnSnap, nil
 }
 
+func (sb *Backend) addParentSeal(chain consensus.ChainReader, header *types.Header) error {
+	number := header.Number.Uint64()
+	logger := sb.logger.New("func", "Backend.addParentSeal()", "number", number)
+
+	// only do this for blocks which start with block 1 as a parent
+	if number <= 1 {
+		return nil
+	}
+
+	// In some cases, "Prepare" may be called before sb.core has moved to the next sequence,
+	// preventing signature aggregation.
+	// This typically happens in round > 0, since round 0 typically hits the "time.Sleep()"
+	// above.
+	// When this happens, loop until sb.core moves to the next sequence, with a limit of 500ms.
+	seq := waitCoreToReachSequence(sb.core, header.Number)
```

Comment:
Is this still strictly necessary? I could be wrong, but finalize should be called once the block is committed and the validator moves onto the next sequence.

Note: this probably doesn't hurt to keep it in.